backend/src/db/migrations/20240502044531_identity-aws-iam-auth.ts

```diff
@@ -0,0 +1,29 @@
+import { Knex } from "knex";
+
+import { TableName } from "../schemas";
+import { createOnUpdateTrigger, dropOnUpdateTrigger } from "../utils";
+
+export async function up(knex: Knex): Promise<void> {
+  if (!(await knex.schema.hasTable(TableName.IdentityAwsIamAuth))) {
+    await knex.schema.createTable(TableName.IdentityAwsIamAuth, (t) => {
+      t.uuid("id", { primaryKey: true }).defaultTo(knex.fn.uuid());
+      t.bigInteger("accessTokenTTL").defaultTo(7200).notNullable();
+      t.bigInteger("accessTokenMaxTTL").defaultTo(7200).notNullable();
+      t.bigInteger("accessTokenNumUsesLimit").defaultTo(0).notNullable();
+      t.jsonb("accessTokenTrustedIps").notNullable();
+      t.timestamps(true, true, true);
+      t.uuid("identityId").notNullable().unique();
+      t.foreign("identityId").references("id").inTable(TableName.Identity).onDelete("CASCADE");
+      t.string("stsEndpoint").notNullable();
```

Comment:
i thought we made it so that users select the region? Or did you figure out something better ?

Reply:
I re-evaluated and realized that it's actually appropriate to allow users to specify the `stsEndpoint` instead of `region` since there're a few unique FIPS-certified endpoints that have a slightly different format being `sts-fips.<region>.amazonaws.com`. Previously, there was an attack-vector consideration in mind but this concern has since been resolved.

It might just be easier to allow people to pass in their STS endpoint and do validation on the list of possible STS endpoints that can be accepted.